exchanges/coinbasepro/coinbasepro_websocket.go

```diff
@@ -46,423 +83,464 @@ func (c *CoinbasePro) WsConnect() error {
-				Price:           wsOrder.Price,
-				Amount:          wsOrder.Size,
-				TriggerPrice:    wsOrder.StopPrice,
-				ExecutedAmount:  wsOrder.Size - wsOrder.RemainingSize,
-				RemainingAmount: wsOrder.RemainingSize,
-				Fee:             wsOrder.TakerFeeRate,
-				Exchange:        c.Name,
-				OrderID:         wsOrder.OrderID,
-				AccountID:       wsOrder.ProfileID,
-				ClientID:        creds.ClientID,
-				Type:            oType,
-				Side:            oSide,
-				Status:          oStatus,
-				AssetType:       a,
-				Date:            ts,
-				Pair:            p,
-			}
-		}
-	case "match", "last_match":
-		var wsOrder wsOrderReceived
-		err := json.Unmarshal(respRaw, &wsOrder)
-		if err != nil {
-			return err
-		}
-		oSide, err := order.StringToOrderSide(wsOrder.Side)
-		if err != nil {
-			c.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: c.Name,
-				Err:      err,
+				return warnString, err
 			}
 		}
-		var p currency.Pair
-		var a asset.Item
-		p, a, err = c.GetRequestFormattedPairAndAssetType(wsOrder.ProductID)
+	case "user":
+		var wsUser []WebsocketOrderDataHolder
+		err := json.Unmarshal(data, &wsUser)
 		if err != nil {
-			return err
-		}
-
-		if wsOrder.UserID != "" {
-			c.Websocket.DataHandler <- &order.Detail{
-				OrderID:   wsOrder.OrderID,
-				Pair:      p,
-				AssetType: a,
-				Trades: []order.TradeHistory{
-					{
-						Price:     wsOrder.Price,
-						Amount:    wsOrder.Size,
-						Exchange:  c.Name,
-						TID:       strconv.FormatInt(wsOrder.TradeID, 10),
-						Side:      oSide,
-						Timestamp: wsOrder.Time,
-						IsMaker:   wsOrder.TakerUserID == "",
-					},
-				},
-			}
-		} else {
-			if !c.IsSaveTradeDataEnabled() {
-				return nil
+			return warnString, err
+		}
+		var sliToSend []order.Detail
+		for i := range wsUser {
+			for j := range wsUser[i].Orders {
+				var oType order.Type
+				oType, err = order.StringToOrderType(wsUser[i].Orders[j].OrderType)
+				if err != nil {
+					return warnString, err
+				}
+				var oSide order.Side
+				oSide, err = order.StringToOrderSide(wsUser[i].Orders[j].OrderSide)
+				if err != nil {
+					return warnString, err
+				}
+				var oStatus order.Status
+				oStatus, err = statusToStandardStatus(wsUser[i].Orders[j].Status)
+				if err != nil {
+					return warnString, err
+				}
+				sliToSend = append(sliToSend, order.Detail{
+					Price:           wsUser[i].Orders[j].AveragePrice,
+					Amount:          wsUser[i].Orders[j].CumulativeQuantity + wsUser[i].Orders[j].LeavesQuantity,
+					ExecutedAmount:  wsUser[i].Orders[j].CumulativeQuantity,
+					RemainingAmount: wsUser[i].Orders[j].LeavesQuantity,
+					Fee:             wsUser[i].Orders[j].TotalFees,
+					Exchange:        c.Name,
+					OrderID:         wsUser[i].Orders[j].OrderID,
+					ClientOrderID:   wsUser[i].Orders[j].ClientOrderID,
+					Type:            oType,
+					Side:            oSide,
+					Status:          oStatus,
+					AssetType:       asset.Spot,
```

Comment:
Given the usage of `asset.All` for "user" and that this can return `productType` of either `SPOT` or `FUTURE` I think you should be returning the appropriate asset type